MJxPrep.js

```diff
@@ -627,8 +630,30 @@ if (window.MJxPrep) {
     funcToPostfix: funcToPostfix
   }
 
-  // A function to allow for custom replacements
   function customReplacements(eqn) {
+    /*
+     * A function to allow for custom replacements.
+     * This function is called *before* wrapping occurs, and should
+     * be used for implementing new functions, but not for doing
+     * cosmetic replacements. The output of this function must still
```

Comment:
I'm confused about what you mean by "cosmetic" change. Isn't everything AsciiMath/MathJax-related cosmetic? 

Is
```js
// Replace trans(...) -> {:(...)^T:}, with parentheses added as necessary
    eqn = replaceFunctionCalls(eqn, 'trans', funcToPostfix('^T') )
```
implementing a new function, or cosmetic?

Reply:
The issue is that our wrapping function has regex that expects things to be syntactically equivalent to the grading library. So, when doing cosmetic changes, they need to be done after the wrapping is complete. I do agree that many of our function implementations are cosmetic... Do you think we can just do wrapping first?

Reply:
Well, it's better if `wrapFuncCalls` comes after anything that replaces traditional function calls `func(args)` with anything else, e.g., postfix notation:

```js
eqn = 'trans(A)'
eqn = wrapVariables(eqn)
eqn = wrapFuncCalls(eqn)
eqn = replaceFunctionCalls(eqn, 'trans', funcToPostfix('^T') )
console.log(eqn)
// {:{:({:A:})^T:}:} // has an extra set of parens and redundant outer `{:...:}`

eqn = 'trans(A)'
eqn = replaceFunctionCalls(eqn, 'trans', funcToPostfix('^T') )
eqn = wrapVariables(eqn)
eqn = wrapFuncCalls(eqn)
console.log(eqn)
// {:{:A:}^{:T:}:}
```
If I move the func/var wrapping after all `replaceFuncCalls` replacements, all of our tests still pass. 

Proposal: offer two custom replacement hooks,
 - `customPreReplacements(eqn)` called before ALL of our replacements except for whitespace cleaning
 - `customPostReplacements(eqn)` called after ALL of our default replacements

I think this is also easier to explain in documentation comments, and leads to more predictable values of the `eqn` argument.

Reply:
Sounds good to me. Want me to implement it in this PR?

Reply:
Sure.